packages/cloudflare/src/entrypoints/image-endpoint.ts

```diff
@@ -1,2 +1,16 @@
-// NOTE: this file is empty on purpose
-// it allows use to offer `imageService: 'compile'`
+import type { APIRoute } from 'astro';
+
+export const prerender = false;
+
+export const GET: APIRoute = (ctx) => {
+	const href = ctx.url.searchParams.get('href');
+	if (!href) {
+		return new Response("Missing 'href' query parameter", { status: 400 });
+	}
+
+	if (!href.startsWith('/_astro')) {
+		return new Response("Invalid 'href' query parameter", { status: 400 });
```

Comment:
I would emit a different error from the previous, otherwise debugging becomes difficult, and we don't tell the user **why** the `href` is invalid.